### DISCUSSION TITLE
Implementation of Read-Only and Disable Features for Forms

### DISCUSSION THREAD

**Discussion by aaronmng** [json-schema-form-element/jsfe#2] [Ideas]
I'm looking for clarification on the capabilities of our form elements. Specifically, I need to know if it's possible to:

1. Set a form or its elements to a read-only state, where fields are pre-populated with default values but cannot be edited by the user.
2. Disable the form or specific elements, preventing any user interaction or editing.

These features are useful for enabling additional functionalities for developers, such as displaying forms filled with data in a non-editable format for viewing purposes.

Could you provide information on whether these features are currently supported, or if there is a plan to include them in future updates?

**Comment by JulianCataldo** [comment]
Thanks for pointing this out.
It's totally planned to expand the UI schema vocabulary, based this on some of this prior art: https://rjsf.netlify.app

Many options like editability, form control size, or even per library-specifics can be added, like Shoelace's `variant`, which can be also leveraged by other lib's or user-made widgets. It's a matter of adding an `ui:foo` + mapping it accordingly in the core lib. and in the UI components themselves.

I'm gonna open this issue (#3) to track tasks.

For the _read-only_ + pre-population feat., I think you might be already able to achieve that.
I'll test when I can it and get back with a code snippet if it works, if not we might add an UI schema field for that.

**Comment by JulianCataldo** [comment]
https://jsfe.js.org/Sink/UiSchemas

<img width="759" alt="Xnapper-2023-11-30-16 40 39" src="https://github.com/json-schema-form-element/jsfe/assets/603498/4f6f8fd8-466d-46bd-bd60-910eddc2ce7e">
